### PR TITLE
add option to make assembly wait till completion

### LINKED ISSUE #28: We should not embrace people to run many GetAssembly requests
In the current Readme sample it says

```
                // you can set your custom wait time
```

We do not want people to go down to x ms. Rather, we should offer a waitUntilFinished() function (and use that in the README examples) that does the while loop with the thread sleeping for us. That is how the Ruby SDK does it for example.

Alternatively, we can offer a waitForCompletion boolean to the assembly.save() function (even though that is weird naming I think). That is how the PHP and Node SDKs do it.

### PR DESCRIPTION
**Pull request by ifedapoolarewaju**
fixes #28

also please advise on any other assembly statuses that should be regarded as "FINISHED". [See here](https://github.com/transloadit/java-sdk/blob/master/src/main/java/com/transloadit/sdk/response/AssemblyResponse.java#L104-L106) cc @tim-kos @Acconut

### CODE REVIEW

**Comment by Acconut** [comment]
In the Go SDK, the wait method is implemented by checking whether the assembly is still uploading/executing instead of testing if it has finished already:
https://github.com/transloadit/go-sdk/blob/19fc9db1b8bd5067f31f8fd9ef4d4f74985a3b5d/wait.go#L24

IIRC, we choose this implementation based on a recommendation by @kvz. Anyways, the advantage is that you don't have to deal with request aborted/assembly error/assembly cancelled statuses and it's more future-proof.

**Comment by ifedapoolarewaju** [comment]
@Acconut I thought there's a state that Assemblies usually are when it is waiting for Tus uploads to begin? I don't remember, but I don't think it is `ASSEMBLY_UPLOADING`. 🤔

**Comment by Acconut** [comment]
> I thought there's a state that Assemblies usually are when it is waiting for Tus uploads to begin? I don't remember, but I don't think it is ASSEMBLY_UPLOADING. 

No, there is no such additional status. The only two states an unfinished assembly can in are ASSEMBLY_UPLOADING and ASSEMBLY_EXECUTING. The tus integration was designed to introduce little changes to the SDKs, so we are also using the ASSEMBLY_UPLOADING state for it.

**Comment by tim-kos** [comment]
Marius is right. Let's change the implementation here and check for `ASSEMBLY_UPLOADING` and `ASSEMBLY_EXECUTING` instead.

File: src/main/java/com/transloadit/sdk/Assembly.java [hunk 1]
```diff
@@ -316,4 +329,26 @@ protected void makeAttempt() throws ProtocolException, IOException {
...
+     * @return {@link AssemblyResponse}
+     * @throws LocalOperationException if something goes wrong while running non-http operations.
+     * @throws RequestException if request to Transloadit server fails.
+     */
+    protected AssemblyResponse waitTillComplete(AssemblyResponse response) throws LocalOperationException, RequestException {
+        try {
+            // wait for assembly to finish executing.
+            while (!response.isFinished()) {
+                Thread.sleep(1000);
+                response = transloadit.getAssemblyByUrl(response.getSslUrl());
```
Comment (tim-kos) [inline src/main/java/com/transloadit/sdk/Assembly.java]: Is it possible at all in Java to listen to Websocket events instead of polling all the time?

Comment (ifedapoolarewaju) [reply src/main/java/com/transloadit/sdk/Assembly.java]: we could use this [socketio client](https://github.com/socketio/socket.io-client-java) yes, the server uses socket.io right?

I would rather create a separate issue for this though.

**Comment by ifedapoolarewaju** [comment]
> Let's change the implementation here and check for ASSEMBLY_UPLOADING and ASSEMBLY_EXECUTING instead.

I have changed as suggested.

**Comment by ifedapoolarewaju** [comment]
@tim-kos @Acconut to merge or not to merge?

**Pull request merged**